### PR TITLE
Add Material 3 `RadioListTile` example and update existing examples

### LINKED ISSUE #101048: Add Material 3 interactive dartpad samples
This is a tracking issue for adding interactive code samples (with `{@tool dartpad}`) to components and widgets which support Material 3. 

We should consider leveraging existing existing code samples by either
- wrapping existing samples with a common widget which shows a toggle to switch the sample to Material 3, at some point, have that be the default (preferred, IMO)
- copy the simplest existing sample and adding `useMaterial3: true` 

For new Material 3-only widgets, such as `NavigationBar`, the [API doc](https://master-api.flutter.dev/flutter/material/NavigationBar-class.html) will only show the Material 3 look.

cc @darrenaustin WDYT?

### Components
- [ ] Buttons
- [x] https://github.com/flutter/flutter/pull/101105
- [ ] #100532
- [x] https://github.com/flutter/flutter/pull/101508
- [x] `NavigationBar`
- [x] https://github.com/flutter/flutter/pull/101345
- [x] https://github.com/flutter/flutter/pull/102823 and https://github.com/flutter/flutter/pull/103962

### Other widgets

### LINKED ISSUE #119700: Update documentation for widgets depended on the `ListTile` widget for Material 3
Since `ListTile` is migrated to Material 3 in https://github.com/flutter/flutter/pull/117965,  widgets depended on the `ListTile` widget need M3 examples and some existing examples clean up.

- [x] `CheckboxListTile` - https://github.com/flutter/flutter/pull/118792
- [x] `SwitchListTile` - https://github.com/flutter/flutter/pull/119714
- [x] `RadioListTile` - https://github.com/flutter/flutter/pull/119716

related to https://github.com/flutter/flutter/issues/101048

### PR DESCRIPTION
**Pull request by TahaTesser**
Fixes https://github.com/flutter/flutter/issues/101048 (part of)
Fixes https://github.com/flutter/flutter/issues/119700 (part of)

### Description
1. Add Material 3 `RadioListTile` example.
2. Update existing examples
3. Add example tests.

### Preview
Screenshot 2023-02-01 at 16 09 29

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by HansMuller — COMMENTED** [review]
This looks good. Please update the examples to useMaterial3:true

File: examples/api/lib/material/radio_list_tile/custom_labeled_radio.0.dart [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            value: value,
+            onChanged: (bool? newValue) {
+              onChanged(newValue!);
+            },
+          ),
+          RichText(
+            text: TextSpan(
+              text: label,
+              style: const TextStyle(
+                color: Colors.blueAccent,
```
Comment (HansMuller) [inline examples/api/lib/material/radio_list_tile/custom_labeled_radio.0.dart:61]: Best to use theme colorScheme colors

**Review by HansMuller — APPROVED** [review]
Nicely done. Glad to see all of the tests!

**Review by pks050505 — COMMENTED** [review]
good

**Pull request merged**